sycl-fusion/passes/kernel-fusion/Builtins.cpp

```diff
@@ -442,59 +425,99 @@ Value *jit_compiler::getGlobalLinearID(IRBuilderBase &Builder,
   return createSPIRVCall(Builder, GetGlobalLinearIDName, {});
 }
 
-Function *jit_compiler::remapBuiltins(Function *F, const NDRange &SrcNDRange,
+static bool isIndexSpaceGetterBuiltin(Function *F) {
+  constexpr std::array<StringLiteral, NumBuiltinsToRemap> BuiltinNames{
+      GetGlobalSizeName,    GetGroupIDName,   GetGlobalOffsetName,
+      GetNumWorkGroupsName, GetLocalSizeName, GetLocalIDName,
+      GetGlobalIDName};
+  const auto Name = F->getName();
+  const auto *Iter = llvm::lower_bound(BuiltinNames, Name);
+  return Iter != BuiltinNames.end() && *Iter == Name;
+}
+
+static bool isSafeToNotRemapBuiltin(Function *F) {
+  constexpr std::size_t NumUnsafeBuiltins{8};
+  // SPIRV builtins with kernel capabilities in alphabetical order.
```

Comment:
Can you document why it's unsafe to remap these?

Reply:
Done!